### PR TITLE
Add support for authentication in Kafka consumers

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Adding support for authentication in the Kafka based consumers

## Type of change

- New feature (non-breaking change which adds functionality)
- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)

## Testing steps

Testes against a RHOSAK instance and a local deployment file

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: ccx_messaging/consumers/consumer.py [hunk 1]
```diff
@@ -62,13 +63,7 @@ def __init__(
...
         downloader,
         engine,
-        group_id=None,
-        incoming_topic=None,
-        bootstrap_servers=None,
-        max_poll_records=None,
-        max_poll_interval_ms=None,
-        heartbeat_interval_ms=None,
-        session_timeout_ms=None,
+        incoming_topic,
```
Comment (tisnik) [inline ccx_messaging/consumers/consumer.py:66]: why was all the other options deleted?

Comment (epapbak) [reply ccx_messaging/consumers/consumer.py:66]: they come in kwargs, so it makes more sense to retrieve them from kwargs.get

Comment (joselsegura) [reply ccx_messaging/consumers/consumer.py:66]: They are default `KafkaConsumer` arguments, they are already "included" as `kwargs`. The only `kwarg`-able argument that is mandatory for us is the `incoming_topic`, so I removed the default value and kept it.

If we want to use those arguments, we just need to put them in the YAML configuration file and they will be passed to the `KafkaConsumer`

**Review by juandspy — APPROVED** [review]

File: ccx_messaging/consumers/consumer.py [hunk 2]
```diff
@@ -77,32 +72,33 @@ def __init__(
...
-            bootstrap_servers,
-            group_id,
+            kwargs.get("bootstrap_servers", None),
+            kwargs.get("group_id", None),
         )
 
-        super().__init__(publisher, downloader, engine)
+        requeuer = kwargs.pop("requeuer", None)
+
+        super().__init__(publisher, downloader, engine, requeuer=requeuer)
```
[the remark this answers is not in the document]
Comment (epapbak) [reply ccx_messaging/consumers/consumer.py:91]: it's supposed to help reprocess a message when something unexpected happened.

**Review by tisnik — APPROVED** [review]
ok

**Pull request merged**